lindi/LindiH5pyFile/LindiH5pyFile.py

```diff
@@ -218,3 +340,60 @@ def _download_file(url: str, filename: str) -> None:
     with urllib.request.urlopen(req) as response:
         with open(filename, "wb") as f:
             f.write(response.read())
+
+
+def _recursive_copy(src_item: Union[h5py.Group, h5py.Dataset], dest: h5py.File, name: str) -> None:
+    if isinstance(src_item, h5py.Group):
+        dst_item = dest.create_group(name)
+        for k, v in src_item.attrs.items():
+            dst_item.attrs[k] = v
+        for k, v in src_item.items():
+            _recursive_copy(v, dest, name=f'{name}/{k}')
+    elif isinstance(src_item, h5py.Dataset):
+        # Let's specially handle the case where the source and dest files
+        # are LindiH5pyFiles with reference file systems as the internal
+        # representation. In this case, we don't need to copy the actual
+        # data because we can copy the reference.
+        if isinstance(src_item.file, LindiH5pyFile) and isinstance(dest, LindiH5pyFile):
+            if src_item.name is None:
+                raise Exception("src_item.name is None")
+            src_item_name = _without_initial_slash(src_item.name)
+            src_zarr_store = src_item.file._zarr_store
+            dst_zarr_store = dest._zarr_store
+            if src_zarr_store is not None and dst_zarr_store is not None:
+                if isinstance(src_zarr_store, LindiReferenceFileSystemStore) and isinstance(dst_zarr_store, LindiReferenceFileSystemStore):
+                    src_rfs = src_zarr_store.rfs
+                    dst_rfs = dst_zarr_store.rfs
+                    src_ref_keys = list(src_rfs['refs'].keys())
+                    for src_ref_key in src_ref_keys:
+                        if src_ref_key.startswith(f'{src_item_name}/'):
+                            dst_ref_key = f'{name}/{src_ref_key[len(src_item_name) + 1:]}'
+                            # Even though it's not expected to be a problem, we
+                            # do a deep copy here because a problem resulting
+                            # from one rfs being modified affecting another
+                            # would be very difficult to debug.
+                            dst_rfs['refs'][dst_ref_key] = _deep_copy(src_rfs['refs'][src_ref_key])
+                    return
+
+        dst_item = dest.create_dataset(name, data=src_item[()])
```

Comment:
```suggestion
        src_item.parent.copy(src_item, dest, name)
```
This should copy all the IO settings and I believe it does not load the dataset into memory.

Reply:
It will also copy attributes by default, so the two lines below can be removed.

If the dataset is a dataset of references, I'm not sure what will happen.

Reply:
I am skipping this change since I'm concerned about creating an infinite recursion. If you feel strongly let me know.

Reply:
OK, that's fine